### DISCUSSION TITLE
How to run "ping" command from within Laravel Sail application

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#51680] [Q&A]
I'm working with Laravel 11 and need to programmatically execute the "ping" command and get the result. I'm using Laravel Sail. But when I attempt to use the `Process` facade I get an error:

> "sh: 1: ping: not found
" // app/Livewire/Site/Forms/Tools/ToolForm.php:48

```php
$result = Process::run('ping -c 3 google.com');

dd($result->errorOutput());
```

How can I achieve this?

**Comment by henzeb** [comment]
Ping needs to be installed in the sail container. It's not installed by default to keep it small

**Reply by sts-ryan-holton** [reply]
How might I go about installing it? `iputils-ping` can't be found when using `sail root-shell` and trying to install the package. Ideally don't want to publish the whole dokcerfile to add one package. What do you recommend to get this resolved?
